warp_drive/training/trainer.py

```diff
@@ -249,6 +227,105 @@ def generate_rollout(self, batch_index):
         torch.cuda.synchronize()
         self.perf_stats.env_step_time += start_event.elapsed_time(end_event) / 1000
 
+    def evaluate_policies(self, batch_index):
+        probabilities = {}
+        for policy in self.policies:
+            probabilities[policy], _ = self.models[policy](
+                batch_index=batch_index, batch_size=self.training_batch_size_per_env
+            )
+
+        # Combine probabilities across policies if there are multiple policies,
+        # yet they are expected to share the same action placeholders
+        if (
+            len(self.policies) > 1
+            and not self.create_separate_placeholders_for_each_policy
+        ):
+            for policy in self.policies:
+                num_action_types[policy] = len(probabilities[policy])
+                probability_shapes[policy] = [
+                    probabilities[policy][action_idx].shape
+                    for action_idx in range(num_action_types[policy])
+                ]
+            assert all_equal(list(num_action_types.values()))
+
+            # Initialize combined_probabilities
+            first_policy = list(probabilities.keys())[0]
+            num_action_types = len(probabilities[first_policy])
+
+            first_action_idx = 0
+            num_envs = probabilities[first_policy][first_action_idx].shape[0]
+            action_dim = probabilities[first_policy][first_action_idx].shape[-1]
```

Comment:
`action_dim` is dependent on `action_type`, may not be defined outside of the loop as a constant, right?

Reply:
yes, correct.

Reply:
This line can be removed